File: TransCoder_tokenized_test_set_functions/cpp/RETURN_PREVIOUS_ELEMENT_IN_AN_EXPANDING_MATRIX.cpp
string findLeft ( string str ) {
  int n = str . length ( );
  while ( n -- ) {
    if ( str [ n ] == 'd' ) {
      str [ n ] = 'c';
      break;
    }
    if ( str [ n ] == 'b' ) {
      str [ n ] = 'a';
      break;
    }
    if ( str [ n ] == 'a' ) str [ n ] = 'b';
    else if ( str [ n ] == 'c' ) str [ n ] = 'd';
  }
  return str;
}

File: results/java_codex/RETURN_PREVIOUS_ELEMENT_IN_AN_EXPANDING_MATRIX.java
string findLeft(string str) {
        int n = str.length();
        while (n > 0) {
            n--;
            if (str[n] == 'd') {
                str[n] = 'c';
                break;
            }
            if (str[n] == 'b') {
                str[n] = 'a';
                break;
            }
            if (str[n] == 'a') str[n] = 'b';
            else if (str[n] == 'c') str[n] = 'd';
        }
        return str;
    }